core/vdbe/execute.rs

```diff
@@ -5872,14 +5872,18 @@ pub fn exec_and(lhs: &OwnedValue, rhs: &OwnedValue) -> OwnedValue {
         | (OwnedValue::Integer(0), _)
         | (_, OwnedValue::Float(0.0))
         | (OwnedValue::Float(0.0), _) => OwnedValue::Integer(0),
-        (OwnedValue::Null, _) | (_, OwnedValue::Null) => OwnedValue::Null,
         (OwnedValue::Text(lhs), OwnedValue::Text(rhs)) => exec_and(
-            &cast_text_to_numeric(lhs.as_str()),
-            &cast_text_to_numeric(rhs.as_str()),
+            &cast_text_to_real(lhs.as_str()),
```

Comment:
just making sure this change to `cast_text_to_real` is intentional since it's not mentioned in the PR description?